modules/core/client/app/init.js

```diff
@@ -19,7 +19,7 @@ angular.module(ApplicationConfiguration.applicationModuleName).run(function ($ro
     if (toState.data && toState.data.roles && toState.data.roles.length > 0) {
       var allowed = false;
       toState.data.roles.forEach(function (role) {
```

Comment:
Probably not best to use Array.forEach if we want to interrupt execution to improve performance.